### PR TITLE
[skip_test] add condition for slash.skipped

### PR DESCRIPTION
**Pull request by eplaut**
This PR allows to set a slash.skipped decorator which applies only for a specific condition

### CODE REVIEW

**Comment by eplaut** [comment]
apparently, `tests/test_skips.py:test_method_with_condition_false` doesn't really checks the test ran.
Can you guide me how to change `tests/test_skips.py:_assert_not_skips`?

File: slash/utils/__init__.py [hunk 1]
```diff
@@ -19,23 +20,29 @@ def repeat(num_repetitions):
...
 
 
-def skipped(thing, reason=None):
+def skipped(thing=NOTHING, reason=None, condition=True):
     """
     A decorator for skipping methods and classes
     """
     from ..core.test import Test
 
+    if thing == NOTHING:
```
Comment (vmalloc) [inline slash/utils/__init__.py]: When comparing sentinels, always use `is`, rather than `==`...

**Comment by vmalloc** [comment]
@eplaut What's the difference between this and [test requirements](http://slash.readthedocs.org/en/master/tour.html#requirements)?

**Comment by eplaut** [comment]
My shame for not understands that `slash.requires` is what I need.

Many thanks!

**Comment by vmalloc** [comment]
It's all good :-) If you need anything else don't hesitate to ask

**Comment by eplaut** [comment]
@vmalloc is there a way to set bunch of tests with a `slash.tag` and to make them run after other test (or another tagged tests) passed?

**Comment by vmalloc** [comment]
Not currently, except for two consecutive runs. What's the use case?

**Comment by eplaut** [comment]
I have bunch of test, some defined as sanity/smoke the others should not run if the sanity tests failed.

right now I change the test name to test_00_sanity so it run first and it set a `g` variable if it pass, and all the other has `@slash.requires` for that variable.

It cause 2 problems:
* The way to sort test is not cleared (and I'm not really sure it will work for any test name)
* It works for one sanity test, it will be more complicated for many of them (possible, but ugly)

**Comment by vmalloc** [comment]
yeah, I understand. This could, in theory, be solved by a hook that can reorder tests prior to executions, based on tags. Currently I don't see any convenient way to solve this.

**Comment by vmalloc** [comment]
Opened https://github.com/slash-testing/slash/issues/362

**Comment by eplaut** [comment]
Thanks!

**Pull request closed without merging**